core/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -71,7 +72,14 @@ protected ClusterHealthResponse newResponse() {
     }
 
     @Override
-    protected void masterOperation(final ClusterHealthRequest request, final ClusterState unusedState, final ActionListener<ClusterHealthResponse> listener) {
+    protected final void masterOperation(ClusterHealthRequest request, ClusterState state, ActionListener<ClusterHealthResponse> listener) throws Exception {
+        logger.warn("attempt to execute a cluster health operation without a task");
```

Comment:
Yes, it was very helpful in debugging test failures caused by calling wrong masterOperation in overrides because the error that is getting thrown is not always logged. So, I decided to leave the logging there, in case somebody else will run into same issue during development in the future.